bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,472 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.resol.handler;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.i18n.LocaleProvider;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.resol.internal.ResolBindingConstants;
+import org.openhab.binding.resol.internal.ResolConfiguration;
+import org.openhab.binding.resol.internal.discovery.ResolDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.resol.vbus.Connection;
+import de.resol.vbus.ConnectionAdapter;
+import de.resol.vbus.Packet;
+import de.resol.vbus.Specification;
+import de.resol.vbus.Specification.PacketFieldValue;
+import de.resol.vbus.SpecificationFile;
+import de.resol.vbus.SpecificationFile.Language;
+import de.resol.vbus.TcpDataSource;
+import de.resol.vbus.TcpDataSourceProvider;
+
+/**
+ * The {@link ResolBridgeHandler} class handles the connection to the
+ * optolink adapter.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+public class ResolBridgeHandler extends BaseBridgeHandler {
```

Comment:
> > The only one where I forgot is the one adding the jar - where it is even the question whether it should be there :-)
> 
> It should be signed-off, too, because you take responsibility, that the library may be used within this binding.

Ah, my main question about the jar got lost in formatting above, so let me put it again:
Can you point me to an example or tutorial on how to work with the external jar? - I was already happy to make it work like this since I Initially even had the Java code copied into the repo - which is probably neither want we want...